cmd/notify-mailer/main.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/mail"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	bmail "github.com/letsencrypt/boulder/mail"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+type mailer struct {
+	clk           clock.Clock
+	log           blog.Logger
+	dbMap         *gorp.DbMap
+	mailer        bmail.Mailer
+	subject       string
+	emailTemplate string
+	destinations  []string
+	checkpoint    interval
+	sleepInterval time.Duration
+}
+
+type interval struct {
+	start int
+	end   int
+}
+
+func (i *interval) ok() error {
+	if i.start < 0 || i.end < 0 {
+		return fmt.Errorf(
+			"interval start (%d) and end (%d) must both be positive integers",
+			i.start, i.end)
+	}
+
+	if i.start > i.end && i.end != 0 {
+		return fmt.Errorf(
+			"interval start value (%d) is greater than end value (%d)",
+			i.start, i.end)
+	}
+
+	return nil
+}
+
+func (m *mailer) ok() error {
+	// Make sure the checkpoint range is OK
+	if checkpointErr := m.checkpoint.ok(); checkpointErr != nil {
+		return checkpointErr
+	}
+
+	// Do not allow a start larger than the # of destinations
+	if m.checkpoint.start > len(m.destinations) {
+		return fmt.Errorf(
+			"interval start value (%d) is greater than number of destinations (%d)",
+			m.checkpoint.start,
+			len(m.destinations))
+	}
+
+	// Do not allow a negative sleep interval
+	if m.sleepInterval < 0 {
+		return fmt.Errorf(
+			"sleep interval (%d) is < 0", m.sleepInterval)
+	}
+
+	return nil
+}
+
+func (m *mailer) run() error {
+	if err := m.ok(); err != nil {
+		return err
+	}
+	// If there is no endpoint specified, use the total # of destinations
+	if m.checkpoint.end == 0 {
+		m.checkpoint.end = len(m.destinations)
+	}
+	for _, dest := range m.destinations[m.checkpoint.start:m.checkpoint.end] {
+		if strings.TrimSpace(dest) == "" {
+			continue
+		}
+		err := m.mailer.SendMail([]string{dest}, m.subject, m.emailTemplate)
+		if err != nil {
+			return err
+		}
+		m.clk.Sleep(m.sleepInterval)
+	}
+	return nil
+}
+
+func main() {
+	from := flag.String("from", "", "From header for emails. Must be a bare email address.")
+	subject := flag.String("subject", "", "Subject of emails")
+	toFile := flag.String("toFile", "", "File containing a list of email addresses to send to, one per file.")
+	bodyFile := flag.String("body", "", "File containing the email body in plain text format.")
+	dryRun := flag.Bool("dryRun", true, "Whether to do a dry run.")
+	sleep := flag.Duration("sleep", 60*time.Second, "How long to sleep between emails.")
+	start := flag.Int("start", 0, "Line of input file to start from.")
+	end := flag.Int("end", 99999999, "Line of input file to end before.")
+	type config struct {
+		NotifyMailer struct {
+			cmd.DBConfig
+			cmd.PasswordConfig
+			cmd.SMTPConfig
+		}
+	}
+	var configFile = flag.String("configFile", "", "File containing a JSON config.")
```

Comment:
Let's make this consistent with our other `--config` flags and just call it `--config`.